test/fuzz/fuzzer_pass_outline_functions_test.cpp

```diff
@@ -0,0 +1,491 @@
+               OpBranch %23
+         %23 = OpLabel
+               OpBranch %21
+         %22 = OpLabel
+               OpBranch %31
+         %31 = OpLabel
+               OpLoopMerge %32 %31 None
+               OpBranchConditional %17 %31 %32
+         %32 = OpLabel
+               OpSelectionMerge %102 None
+               OpBranchConditional %17 %34 %35
+         %34 = OpLabel
+               OpBranch %102
+         %35 = OpLabel
+               OpBranch %102
+        %102 = OpLabel
+        %103 = OpPhi %12 %18 %34 %18 %35
+               OpBranch %104
+        %104 = OpLabel
+               OpBranch %33
+         %33 = OpLabel
+         %42 = OpPhi %12 %103 %104 %19 %33
+               OpLoopMerge %36 %33 None
+               OpBranchConditional %17 %36 %33
+         %36 = OpLabel
+               OpReturn
+         %37 = OpLabel
+               OpLoopMerge %38 %39 None
+               OpBranch %40
+         %40 = OpLabel
+               OpBranchConditional %17 %41 %38
+         %41 = OpLabel
+               OpBranch %39
+         %39 = OpLabel
+               OpBranch %37
+         %38 = OpLabel
+               OpReturn
+               OpFunctionEnd
+)";
+
+  ASSERT_TRUE(IsEqual(env, after_adjustments, context.get()));
+}
+
+TEST(FuzzerPassOutlineFunctionsTest, ExitBlock) {
+  const auto env = SPV_ENV_UNIVERSAL_1_5;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  FactManager fact_manager;
+  spvtools::ValidatorOptions validator_options;
+  TransformationContext transformation_context(&fact_manager,
+                                               validator_options);
+
+  PseudoRandomGenerator prng(0);
+  FuzzerContext fuzzer_context(&prng, 100);
+  protobufs::TransformationSequence transformation_sequence;
+
+  FuzzerPassOutlineFunctions fuzzer_pass(context.get(), &transformation_context,
+                                         &fuzzer_context,
+                                         &transformation_sequence);
+
+  // Block 39 is not a merge block, so it is already suitable.
+  auto suitable_exit_block = fuzzer_pass.MaybeGetExitBlockSuitableForOutlining(
+      context->get_instr_block(39));
+  ASSERT_TRUE(suitable_exit_block);
+  ASSERT_TRUE(suitable_exit_block->GetLabel()->result_id() == 39);
+
+  // The following are merge blocks and, thus, they will need to be split.
+
+  // Block 22
+  suitable_exit_block = fuzzer_pass.MaybeGetExitBlockSuitableForOutlining(
+      context->get_instr_block(22));
+  ASSERT_TRUE(suitable_exit_block);
+  ASSERT_TRUE(suitable_exit_block->GetLabel()->result_id() == 100);
+
+  // Block 27
+  suitable_exit_block = fuzzer_pass.MaybeGetExitBlockSuitableForOutlining(
+      context->get_instr_block(27));
+  ASSERT_TRUE(suitable_exit_block);
+  ASSERT_TRUE(suitable_exit_block->GetLabel()->result_id() == 101);
+}
+}  // namespace
+}  // namespace fuzz
+}  // namespace spvtools
```

Comment:
Please have a new line at the end of the file.

Reply:
Fixed